back/engines/commercial/insights/app/controllers/insights/web_api/v1/detected_categories_controller.rb

```diff
@@ -4,31 +4,24 @@ module Insights
   module WebApi::V1
     class DetectedCategoriesController < ::ApplicationController
       skip_after_action :verify_policy_scoped, only: :index # The view is authorized instead.
-      after_action :verify_authorized
 
       def index
-        render json: dummy_detected_categories
+        render json: Insights::WebApi::V1::DetectedCategorySerializer.new(detected_categories).serialized_json
       end
 
       private
 
       def view
         @view ||= authorize(
-          View.includes(:categories).find(params.require(:view_id)),
+          View.includes(:detected_categories, :categories).find(params.require(:view_id)),
           :show?
         )
       end
 
-      def dummy_detected_categories
-        {
-          data: {
-            names: hardcoded_category_names - view.categories.pluck(:name)
-          }
-        }
-      end
 
-      def hardcoded_category_names
-        %w[housing rental city home rent tax people income vancouver build new need affordable cost live building unit provide money help low family support pay house]
+      def detected_categories
+        category_names = view.categories.map { |category| category.name }
+        view.detected_categories.select { |detected_category| !category_names.include?(detected_category.name) }.take(10)
```

Comment:
[with Rubocop's voice]

```suggestion
        view.detected_categories.reject { |detected_category| category_names.include?(detected_category.name) }.take(10)
```